frontend/workflows/projectSelector/src/helpers.tsx

```diff
@@ -47,6 +48,12 @@ const updateGroupstate = (
   return newState;
 };
 
+const updateHidden = (state: State, group: Group, project: string, hidden: boolean): State => {
+  const newState = { ...state };
+  newState[group][project].hidden = hidden;
```

Comment:
if it's derived you'll remove it from the state entirely and only use it when rendering. should not be in both places.

Reply:
yeah I did it like this so that we'd have a way to know whether upstream/downstream project data should be passed to cards (ie "if checked but not hidden"). 🤔  perhaps I can just do one - use the `deriveHiddenStatus` to only update the hidden state and then check the `hidden` field to decide whether to render

Reply:
the effect that will update the parent context is in the component, so it will have access to which projects are hidden or not when it updates the parent context

Reply:
updated PR to have the derived computation in the project-selector component. i also tested this on your `dash-data-flow` branch and it worked nicely. in the new useEffect hook to update selected projects, I used the `derivedState` value.